src/sql/workbench/contrib/dataExplorer/browser/dataExplorerExtensionPoint.ts

```diff
@@ -125,6 +155,41 @@ export class DataExplorerContainerExtensionHandler implements IWorkbenchContribu
 		});
 	}
 
+	private registerExtensionAction(actionOptions: IDataExplorerActionOptions): IDisposable {
```

Comment:
What is all this logic for? If you're only bringing back the new deployment item then is all this really necessary? It seems like you could just call MenuRegister.appendMenuItems directly with a menu whose ID is `MenuId.ViewContainerTitle` and when of `ContextKeyEqualsExpr.create('viewContainer', VIEWLET_ID)` here. As far as I know we only have the single contribution in this menu anyways so that should be fine for now.

Reply:
This is how VSCode does now, so I am following their pattern. The Views sub menu is an example.

Reply:
I would assume they do it because they have a bunch of different actions being contributed. We don't need most of this logic though so in the interest of keeping things simple I think it would be best to just add the menu action once directly.